pytest/tests/replay/replay.py

```diff
@@ -0,0 +1,177 @@
+    print('New genesis ready.')
+    print('Please launch your localnet node now.')
+    print('Hint: nearup run localnet --home %s --num-nodes 1' % hint_dir)
+
+def convert_snack_case_to_camel_case(s):
+    words = s.split('_')
+    return words[0] + ''.join(w.title() for w in words[1:])
+
+def convert_json_rust_instance_to_py_object(ordered_dict_instance, class_name):
+    ans = class_name()
+    for attr_key, attr_value in ordered_dict_instance.items():
+        setattr(ans, convert_snack_case_to_camel_case(attr_key), attr_value)
+    return ans
+
+def convert_transaction_type_string_to_class(name):
+    tx_class_by_type = {
+        "CreateAccount": CreateAccount,
+        "DeleteAccount": DeleteAccount,
+        "DeployContract": DeployContract,
+        "FunctionCall": FunctionCall,
+        "Transfer": Transfer,
+        "Stake": Stake,
+        "AddKey": AddKey,
+        "DeleteKey": DeleteKey,
+    }
+    if name in tx_class_by_type:
+        return tx_class_by_type[name]
+    raise ValueError('Unknown tx type: %s' % name)
+
+def convert_json_public_key_to_py_public_key(json_public_key):
+    pk_str = json_public_key.split(':')[1] if ':' in json_public_key else json_public_key
+    ans = PublicKey()
+    ans.keyType = 0
+    ans.data = base58.b58decode(pk_str.encode('ascii'))
+    return ans
+
+'''
+Get tx fields read from json ready to be serialized.
+See `pytest/lib/messages/` for tx schemas desired.
+'''
+def fix_json_fields_by_tx_type(py_tx, tx_type):
+    if tx_type == "CreateAccount":
+        pass
+    elif tx_type == "DeleteAccount":
+        pass
+    elif tx_type == "DeployContract":
+        py_tx.code = base64.b64decode(py_tx.code.encode('ascii'))
+    elif tx_type == "FunctionCall":
+        py_tx.args = base64.b64decode(py_tx.args.encode('ascii'))
+        py_tx.deposit = int(py_tx.deposit)
+    elif tx_type == "Transfer":
+        py_tx.deposit = int(py_tx.deposit)
+    elif tx_type == "Stake":
+        py_tx.stake = int(py_tx.stake)
+        py_tx.publicKey = convert_json_public_key_to_py_public_key(py_tx.publicKey)
+    elif tx_type == "AddKey":
+        raise ValueError('Unsupported tx type: %s' % tx_type)
+    elif tx_type == "DeleteKey":
+        py_tx.publicKey = convert_json_public_key_to_py_public_key(py_tx.publicKey)
+    else:
+        raise ValueError('Unknown tx type: %s' % tx_type)
+    return py_tx
+
+def convert_json_action_to_py_action(action_dict):
+    assert(len(action_dict) == 1)
+    for tx_type, value in action_dict.items():
+        contents = convert_json_rust_instance_to_py_object(value, convert_transaction_type_string_to_class(tx_type))
+        contents = fix_json_fields_by_tx_type(contents, tx_type)
+        action = Action()
+        action.enum = tx_type[0].lower() + tx_type[1:]
+        setattr(action, action.enum, contents)
+        return action
+
+def send_resigned_transactions(tx_path, home_dir):
+    with open(os.path.join(home_dir, 'node0/', 'node_key.json'), 'r') as fin:
```

Comment:
btw - this node0 is a little bit surprising here. (as you should be looking in home_dir directly).

Reply:
I thought through this again today. The `generate` section places its json output under `node0` directory, because `nearup` expects the files to be under `node0` directory for localnet with one node. The `send` section here merely replicates that pattern so that the two sections have the same path semantics. 

Let me know if this is an issue.